### ISSUE TITLE
Several global constants lost in 5.2.1

### ISSUE THREAD

**Issue by davidlbaird** [pytest-dev/pyfakefs#809]
Upon importing pyfakefs 5.2.1, we see numerous errors due to lost global constants in fake_filesystem.py.

```
No attribute 'PERM_ALL' 
No attribute 'PERM_DEF'
No attribute 'PERM_DEF_FILE'
No attribute 'PERM_EXE'
No attribute 'PERM_READ'
No attribute 'PERM_WRITE'
No attribute 'USER_ID'
```

Please re-define them, for backward compatibiltiy.

Thanks.

**Comment by davidlbaird** [comment]
Instead of USER_ID, I think a get_uid function should be added.

**Comment by mrbean-bremen** [comment]
Thanks! Sorry for that, that had been due to the refactoring I made for 5.2.0 (to make pytype a bit faster...).
I think I had omitted the `PERM_` constants intentionally (moved them to `helpers`), not being aware that they may be used from outside. Bad guess...
Same for `USER_ID`, which can be set using `set_uid` and get using `os.get_uid`, but I guess `get_uid` would be a good idea for symmetry.
Will get at this tonight.

**Comment by davidlbaird** [comment]
`getuid` is conspicuously missing from `FakeOsModule`. I also found that simply setting USER_ID = helpers.USER_ID in fake_filesystem.py does not capture changes to the value after calling `set_uid`.

**Comment by mrbean-bremen** [comment]
> getuid is conspicuously missing from FakeOsModule

Hm, I was sure that is there, but you are right of course. Will add this.

> I also found that simply setting USER_ID = helpers.USER_ID in fake_filesystem.py does not capture changes to the value after calling set_uid

Also true. This will only correctly work for constants, not for mtuable POD globals, as otherwise always the local copy is used. This can only be fixed by moving the globals back, or adapting the user code to use `get_uid` instead, after it is added (or `os.getuid` after it is fixed).

Probably have to move the constants back...

**Comment by davidlbaird** [comment]
Adding this also works, but perhaps accessing the value through os.getuid (and setting, too? look through the os module's functions)

```
def __getattr__(name):
    if name == 'USER_ID':
        return helpers.USER_ID
    raise AttributeError("No attribute %r." % name)
```

**Comment by mrbean-bremen** [comment]
> Adding this also works

Ah, thanks - I forgot about this! I had already used that elsewhere... This is read-only, there is no possibility for a module-level property setter as far as I know.
Yes, I will add this for backwards compatibilty to read the globals - setting them has to be done via `set_uid`, which was available and documented before. I will add this to the PR, and you may have a look at it.

Thanks for the help!

**Comment by mrbean-bremen** [comment]
Please check if the changes are sufficient for you - if yes, I will merge them and probably make another patch release shortly.

**Comment by davidlbaird** [comment]
I looked at the changes, and they will work. Thanks!

**Comment by mrbean-bremen** [comment]
Fixed in just released version 5.2.2, closing.
